### ISSUE TITLE
Random failure creating windows node pool

### ISSUE THREAD

**Issue by Al77056** [terraform-google-modules/terraform-google-kubernetes-engine#1383]
### TL;DR

Additional node pools are created in random order. If the list contains a windows node pool and the "remove_default_node_pool" variable is set to true, randomly the windows node pool could be come the first one to be created, and result in error such as the following:
error creating NodePool: googleapi: Error 400: WINDOWS_SAC and WINDOWS_LTSC image families require at least one other Linux node pool (e.g. COS_CONTAINERD, COS, UBUNTU) in the cluster., badRequest

### Expected behavior

Additional node pools should be created in the order it is specified in variable.node_pools.

### Observed behavior

Additional node pools are created in random order.

### Terraform Configuration

```hcl
example node pool configuration

node_pools = [
  {
    name                      = "cpu58p"
    machine_type              = "c2-standard-60"
    node_locations            = "us-central1-f"
    min_count                 = 0
    max_count                 = 50
    auto_upgrade              = false
  },
  {
    name                      = "cpu110"
    machine_type              = "c2d-standard-112"
    node_locations            = "us-central1-f"
    spot                      = false
    min_count                 = 0
    max_count                 = 50
    auto_upgrade              = false
  },
  {
    name                      = "cpu4w"
    machine_type              = "n1-standard-4"
    min_count                 = 0
    max_count                 = 50
    image_type                = "windows_ltsc_containerd"
    disk_type                 = "pd-ssd"
    auto_upgrade              = false
    min_cpu_platform          = "Intel Skylake"
  },
]
```

### Terraform Version

```sh
Terraform v1.2.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.33.0
+ provider registry.terraform.io/hashicorp/google-beta v4.34.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.13.1
+ provider registry.terraform.io/hashicorp/random v3.3.2
```

### Additional information

I am guessing the problem is line 48 in main.tf, var.node_pools is already a list, there's no need to convert it to a set.

  node_pool_names = [for np in toset(var.node_pools) : np.name]

**Comment by bharathkkb** [comment]
Thanks for the report @Al77056 . I am not sure if conversion of var.node_pools to a set is a factor but rather it is Terraform trying to create n node pools in parallel. One way to sequence this in the module could be to split the nodepools into two resource groups - one for windows and one for other. This will then allow the windows group to depend on the others group.

**Comment by Al77056** [comment]
hi @bharathkkb , the order of the node pool play in a main role as in practice we create dozens of node pools, and the windows node pools are the last one in the list. So a not-so-practical workaround is to run the plan several times until a plan is generated with acceptable order and then specify the parallelism during apply to avoid the windows node pools to be created in the first round of node pools.

I have tested change of line 48 and 49 in main.tf into the following in my local area and it creates the node pools in the order specified in the variables, which works in our case where a relative large number of node pools are created with windows node pools specified in the end of the list. A better fix is probably introduce a dependency from the windows node pool to a linux node pool, which will force the windows node pools to be created after the linux node pool.

  node_pool_names = [for np in var.node_pools : np.name]
  node_pools      = zipmap(local.node_pool_names, var.node_pools)

**Comment by g-awmalik** [comment]
PR merged for this feature.
